src/clj/fluree/db/connection/config.cljc

```diff
@@ -61,18 +61,78 @@
   (and (storage? node)
        (contains? node conn-vocab/ipfs-endpoint)))
 
+(defn env-var-datatype?
+  [node]
+  (-> node util/get-datatype (= conn-vocab/env-var-datatype)))
+
+(defn get-value
+  [node]
+  (let [v (util/get-value node)]
+    (if (env-var-datatype? node)
+      #?(:clj (or (System/getenv v)
+                  (throw (ex-info (str "Environment variable " v " unset.")
+                                  {:status 400 :error :db/missing-env-var})))
+         :cljs nil) ; TODO: Support environment variable overrides in cljs
```

Comment:
I think we should throw in this case rather than emit `nil` just to be on the safe side.